src/main/java/org/folio/rest/impl/UsersAPI.java

```diff
@@ -468,30 +482,89 @@ public void postUsersOutboxProcess(Map<String, String> okapiHeaders, Handler<Asy
-      if (isDuplicateUsernameError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "User with this username already exists")));
-        return;
-      }
-      if (isDuplicateBarcodeError(reply)) {
-        asyncResultHandler.handle(
-          succeededFuture(PutUsersByUserIdResponse
-            .respond400WithTextPlain(
-              "This barcode has already been taken")));
-        return;
-      }
-      logger.debug("Save successful");
-      asyncResultHandler.handle(reply);
-    });
+
+    pgClient.withTrans(conn -> usersService.getUserById(conn, entity.getId())
+        .compose(userFromStorage -> {
+          if (userFromStorage == null) {
+            return succeededFuture(PutUsersByUserIdResponse.respond404WithTextPlain(entity.getId()));
+          }
+          boolean isConsortiaFieldsUpdated = isConsortiumUserFieldsUpdated(entity, userFromStorage);
+
+          return usersService.updateUser(conn, entity)
+            .compose(user -> {
+              if (isConsortiaFieldsUpdated) {
+                return userOutboxService.saveUserOutboxLog(conn, user, UserEvent.Action.EDIT, okapiHeaders);
+              }
+              return Future.succeededFuture();
+            })
+            .map(aVoid -> PutUsersByUserIdResponse.respond204())
+            .map(Response.class::cast);
+        })
+      .onComplete(reply -> {
+
+        if (reply.cause() != null) {
+          handleUpdateUserFailures(asyncResultHandler, reply);
+          return;
+        }
+
+        userOutboxService.processOutboxEventLogs(vertxContext.owner(), okapiHeaders);
+        asyncResultHandler.handle(reply);
+      }));
+  }
+
+  private void handleUpdateUserFailures(Handler<AsyncResult<Response>> asyncResultHandler, AsyncResult<Response> reply) {
+    String errorMessage = reply.cause().getMessage();
+    if (isDuplicateUsernameError(errorMessage)) {
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "User with this username already exists")));
+      return;
+    }
+
+    if (isDuplicateBarcodeError(errorMessage)) {
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(
+            "This barcode has already been taken")));
+      return;
+    }
+
+    if (reply.cause() instanceof PgException) {
+      asyncResultHandler.handle(
+        succeededFuture(PutUsersByUserIdResponse
+          .respond400WithTextPlain(((PgException) reply.cause()).getDetail())));
+      return;
+    }
+
+    asyncResultHandler.handle(
+      succeededFuture(PutUsersByUserIdResponse
+        .respond400WithTextPlain(errorMessage))
+    );
+  }
+
+  private boolean isConsortiumUserFieldsUpdated(User updatedUser, User userFromStorage) {
+    if (ObjectUtils.notEqual(userFromStorage.getUsername(), updatedUser.getUsername())) {
```

Comment:
Please follow the logging standards for e.g. start of the method a log and end of the method info log before the return statements etc..

Reply:
Added log.